PinwheelSDK/Classes/Pinwheel.swift

```diff
@@ -116,27 +178,79 @@ private func getScript(token: String, initializationTime: Int64) -> String {
+                            }
+                            break;
+                        case "\(PinwheelEventType.login.rawValue)":
+                            if (window.webkit.messageHandlers.\(PinwheelEventHandler.loginEventHandler.rawValue)) {
+                                window.webkit.messageHandlers.\(PinwheelEventHandler.loginEventHandler.rawValue).postMessage(JSON.stringify(event.data));
+                            }
+                            break;
+                        case "\(PinwheelEventType.inputAmount.rawValue)":
+                            if (window.webkit.messageHandlers.\(PinwheelEventHandler.inputAmountEventHandler.rawValue)) {
+                                window.webkit.messageHandlers.\(PinwheelEventHandler.inputAmountEventHandler.rawValue).postMessage(JSON.stringify(event.data));
+                            }
+                            break;
+
+                        case "\(PinwheelEventType.exit.rawValue)":
+                            if (window.webkit.messageHandlers.\(PinwheelEventHandler.exitEventHandler.rawValue)) {
+                                window.webkit.messageHandlers.\(PinwheelEventHandler.exitEventHandler.rawValue).postMessage(JSON.stringify(event.data));
+                            }
+                            break;
+                        case "\(PinwheelEventType.success.rawValue)":
+                            if (window.webkit.messageHandlers.\(PinwheelEventHandler.successEventHandler.rawValue)) {
+                                window.webkit.messageHandlers.\(PinwheelEventHandler.successEventHandler.rawValue).postMessage(JSON.stringify(event.data));
+                            }
+                            break;
+                        case "\(PinwheelEventType.error.rawValue)":
+                            if (window.webkit.messageHandlers.\(PinwheelEventHandler.errorEventHandler.rawValue)) {
+                                window.webkit.messageHandlers.\(PinwheelEventHandler.errorEventHandler.rawValue).postMessage(JSON.stringify(event.data));
+                            }
+                            break;
+                    }
                 }
             }
         });
-
         window.postMessage(
           {
             type: 'PINWHEEL_INIT',
             payload: {
-              sdk: "ios",
-              fullScreen: true,
-              linkToken: "\(token)",
-              uniqueUserId: uuid,
-              initializationTimestamp: \(initializationTime)
+                sdk: "ios",
+                fullScreen: true,
+                linkToken: "\(token)",
+                uniqueUserId: uuid,
+                initializationTimestamp: \(initializationTime),
+                version: {
+                    major: \(version[0]),
+                    minor: \(version.count > 1 ? version[1] : "0"),
+                    patch: \(version.count > 2 ? version[2] : "0")
+                },
+                initializationOptions: {
```

Comment:
Do you require all event listeners be specified by the user? All other SDKs make them optional which I think is the better experience.

Reply:
I added some empty extension implementations to make them optional to implement for the user. There's probably a way to reflect that here—is it okay to leave that for a minor version update?

Reply:
It's probably better to comment this out in that case vs. send inaccurate data.

Reply:
Ah, got it. I mistakenly thought I needed those to receive the callbacks. Removed!

Reply:
Oh yeah no the purpose of these is to log for analytics purposes which of the optional callbacks the user passed in a listener for so we know which are being used.